src/Enzyme.jl

```diff
@@ -133,6 +133,7 @@ Base.convert(::Type{API.CDerivativeMode}, ::ForwardMode) = API.DEM_ForwardMode
 function guess_activity end
 
 mutable struct EnzymeContext
+    world::UInt64
```

Comment:
this should already be accessible everywhere gutils is available per enzyme_extract_world(fn) or something

Reply:
This is meant as a proof of concept to demonstrate that yes we can get the world we are compiling from everywhere. 

Using a string attribute is fine, but is also a blocker for caching the IR for later use, so I wanted to see if we can use this to avoid adding ephemeral data to potentially cacheable inputs.